keras_cv/layers/preprocessing/random_crop_test.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+
+import numpy as np
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers.preprocessing.random_crop import RandomCrop
+
+
+class RandomCropTest(tf.test.TestCase, parameterized.TestCase):
+    def _run_test(self, expected_height, expected_width):
+        np.random.seed(1337)
+        num_samples = 2
+        orig_height = 5
+        orig_width = 8
+        channels = 3
+        input = tf.random.uniform(
+            shape=[num_samples, orig_height, orig_width, channels],
+        )
+        layer = RandomCrop(expected_height, expected_width)
+        actual_output = layer(input)
+        expected_output = tf.random.uniform(
+            shape=(
+                num_samples,
+                expected_height,
+                expected_width,
+                channels,
+            ),
+        )
+        self.assertAllEqual(expected_output.shape, actual_output.shape)
+
+    def test_input_smaller_than_crop_box(self):
+        np.random.seed(1337)
+        height, width = 10, 8
+        inp = np.random.random((12, 3, 3, 3))
+        layer = RandomCrop(height, width)
+        actual_output = layer(inp)
+        # In this case, output should equal resizing with crop_to_aspect
+        # ratio.
+        resizing_layer = tf.keras.layers.Resizing(height, width)
+        expected_output = resizing_layer(inp)
+        self.assertAllEqual(expected_output, actual_output)
+
+    def test_training_with_mock(self):
+        np.random.seed(1337)
+        height, width = 3, 4
+        height_offset = np.random.randint(low=0, high=3)
+        width_offset = np.random.randint(low=0, high=5)
+        mock_offset = [height_offset, width_offset]
+        layer = RandomCrop(height, width)
+        with unittest.mock.patch.object(
+            layer._random_generator,
+            "random_uniform",
+            return_value=mock_offset,
+        ):
+            inp = np.random.random((12, 5, 8, 3))
+            actual_output = layer(inp, training=True)
+            expected_output = inp[
+                :,
+                height_offset : (height_offset + height),
+                width_offset : (width_offset + width),
+                :,
+            ]
+            self.assertAllClose(expected_output, actual_output)
+
+    @parameterized.named_parameters(
```

Comment:
It seems like all of these tests which delegate to `self._run_test` could just be expressed as parameterizations of `_run_test`, and that test could be called `test_output_shape`

Reply:
done!